src/OpenTelemetry.Api/Internal/ActivityHelperExtensions.cs

```diff
@@ -42,19 +40,40 @@ public static bool TryGetStatus(this Activity activity, out StatusCode statusCod
         {
             Debug.Assert(activity != null, "Activity should not be null");
 
-            ActivityStatusTagEnumerator state = default;
+            StatusCode? tempStatusCode = null;
+            string tempStatusDescription = null;
 
-            ActivityTagsEnumeratorFactory<ActivityStatusTagEnumerator>.Enumerate(activity, ref state, null);
+            foreach (ref readonly var tag in activity.EnumerateTagObjects())
+            {
+                if (tag.Key == SpanAttributeConstants.StatusCodeKey)
+                {
+                    tempStatusCode = StatusHelper.GetStatusCodeForTagValue(tag.Value as string);
+                }
+                else if (tag.Key == SpanAttributeConstants.StatusDescriptionKey)
+                {
+                    tempStatusDescription = tag.Value as string;
+                }
+                else
+                {
+                    continue;
+                }
+
+                if (tempStatusCode.HasValue && tempStatusDescription != null)
+                {
+                    goto Done;
```

Comment:
`goto` ... whoa.. I mean cool?

Couldn't you just return here setting `statusCode` and `statusDescription` and initializing them at the top too?

Reply:
Poor `goto` 🤣 Updated.